apiserver/facades/agent/provisioner/provisioninginfo.go

```diff
@@ -579,15 +583,24 @@ func (api *ProvisionerAPI) availableImageMetadata(
 }
 
 // constructImageConstraint returns model-specific criteria used to look for image metadata.
-func (api *ProvisionerAPI) constructImageConstraint(m *state.Machine, env environs.Environ) (*imagemetadata.ImageConstraint, error) {
+func (api *ProvisionerAPI) constructImageConstraint(
+	ctx context.Context,
+	m *state.Machine,
+	env environs.Environ,
```

Comment:
todo: Investigate if each of these methods still require an environ if model config is passed in.